tableaudocumentapi/datasource.py

```diff
@@ -107,3 +117,17 @@ def version(self):
     @property
     def connections(self):
         return self._connections
+
+    ###########
+    # fields
+    ###########
+    @property
+    def columns(self):
+        if not self._columns:
+            self._columns = self._get_all_columns()
+        return self._columns
+
+    def _get_all_columns(self):
+        column_objects = (_mapping_from_xml(self._datasourceTree, xml)
+                          for xml in self._datasourceTree.findall('.//column'))
+        return {k: v for k, v in column_objects}
```

Comment:
I think if this were an ordereddict it would match the document order (and thus the UI?)